crates/uv-auth/src/middleware.rs

```diff
@@ -42,34 +76,39 @@ impl Middleware for AuthMiddleware {
     ) -> reqwest_middleware::Result<Response> {
         let url = request.url().clone();
 
-        // If the request already has an authorization header, we don't need to do anything.
-        // This gives in-URL credentials precedence over the netrc file.
+        // Stash any authentication attached to this request for future requests
+        // in the same realm.
+        self.authentication_store()
+            .set_default_from_request(&request);
```

Comment:
This should mean we do not need to "pre-populate" the store from the URLs beforehand. I wonder if we still should?

This should be the only behavioral change here. I don't think it's user-facing, but we'll extract and store authentication from the headers.